trunk/SUAVE/Components/Energy/Networks/Propulsor_Surrogate.py

```diff
@@ -32,159 +34,322 @@ class Propulsor_Surrogate(Propulsor):
+        else:
+            sfc = sfc_surrogate.predict(cond)
+            thr = thr_surrogate.predict(cond)
 
-        
+        sfc = sfc*self.sfc_input_scale*self.sfc_anchor_scale
+        thr = thr*self.thrust_input_scale*self.thrust_anchor_scale
+       
         F    = thr
         mdot = thr*sfc*self.number_of_engines
-        
+       
         # Save the output
         results = Data()
-        results.thrust_force_vector = self.number_of_engines * F * [np.cos(self.thrust_angle),0,-np.sin(self.thrust_angle)]     
+        results.thrust_force_vector = self.number_of_engines * F * [np.cos(self.thrust_angle),0,-np.sin(self.thrust_angle)]
         results.vehicle_mass_rate   = mdot
-    
+   
         return results          
     
     def build_surrogate(self):
         """ Build a surrogate. Multiple options for models are available including:
             -Gaussian Processes
             -KNN
             -SVR
-    
+            
             Assumptions:
             None
-    
+            
             Source:
             N/A
-    
+            
             Inputs:
             state [state()]
-    
+            
             Outputs:
             self.sfc_surrogate    [fun()]
             self.thrust_surrogate [fun()]
-    
+            
             Properties Used:
             Defaulted values
         """          
-        
+       
         # file name to look for
         file_name = self.input_file
-        
+       
         # Load the CSV file
         my_data = np.genfromtxt(file_name, delimiter=',')
-        
+       
         # Remove the header line
         my_data = np.delete(my_data,np.s_[0],axis=0)
-        
+       
         # Clean up to remove redundant lines
         b = np.ascontiguousarray(my_data).view(np.dtype((np.void, my_data.dtype.itemsize * my_data.shape[1])))
         _, idx = np.unique(b, return_index=True)
-        
+       
         my_data = my_data[idx]                
-                
-    
+               
+   
         xy  = my_data[:,:3] # Altitude, Mach, Throttle
         thr = np.transpose(np.atleast_2d(my_data[:,3])) # Thrust
         sfc = np.transpose(np.atleast_2d(my_data[:,4]))  # SFC
         
+        self.altitude_input_scale = np.max(xy[:,0])
+        self.thrust_input_scale   = np.max(thr)
+        self.sfc_input_scale      = np.max(sfc)
         
+        # normalize for better surrogate performance
+        xy[:,0] /= self.altitude_input_scale
+        thr     /= self.thrust_input_scale
+        sfc     /= self.sfc_input_scale
+       
+       
         # Pick the type of process
-        if self.surrogate_type  == 'gaussian': 
-            gp_kernel_RQ = RationalQuadratic(length_scale=1.0, alpha=1.0)
-            regr_sfc = gaussian_process.GaussianProcessRegressor(kernel=gp_kernel_RQ,normalize_y=True)
-            regr_thr = gaussian_process.GaussianProcessRegressor(kernel=gp_kernel_RQ)      
+        if self.surrogate_type  == 'gaussian':
+            gp_kernel = Matern()
```

Comment:
Just want to check. Have you checked the matern kernel for behavior?

Reply:
It performed the best out of several kernels I tested on a low bypass turbofan data set from Raymer. I also spot checked it in a couple places/slices with my linear data set, and that looked good too.